gui/GUI.java

```diff
@@ -0,0 +1,38 @@
+package gui;
+
+import javafx.scene.*;
+import javafx.scene.canvas.Canvas;
+import javafx.stage.Stage;
+
+public class GUI {
+
+	private Scene scene;
+	private Canvas canvas;
+	private static GUI instance = null;
+
+	private GUI() {
+		// private so there's only one
+	}
+
+	public static GUI getInstance() {
+		if (instance == null)
+			instance = GUIBuilder.Build(new GUI()); //this one
+		return instance;
+	}
+
+	public void setScene(Scene scene) {
+		this.scene = scene;
+	}
+	
+	public void setCanvas(Canvas canvas) {
+		this.canvas = canvas;
+	}
+
+	public void show(Stage stage) {
```

Comment:
Could use a better method name-- currently unclear as to what this does specifically.